src/wagtail_live/adapters/telegram/receiver.py

```diff
@@ -0,0 +1,273 @@
+        except self.model.DoesNotExist:
+            return
+
+        message_id = self.get_message_id_from_message(message=message)
+        live_post = live_page.get_live_post_by_message_id(message_id=message_id)
+
+        files = self.get_message_files(message=message)
+        self.process_files(live_post=live_post.value, files=files)
+
+        live_page.update_live_post(live_post=live_post)
+
+    def handle_bot_command(self, message):
+        """Handles the following bot commands:
+        /get_chat_id: returns the id of the current chat.
+        """
+
+        command = message["entities"][0]
+        start = command["offset"]
+        end = start + command["length"]
+        command_text = message["text"][start:end]
+
+        if command_text == "/get_chat_id":
+            chat_id = self.get_channel_id_from_message(message=message)
+            response = requests.get(
+                get_base_telegram_url() + "sendMessage",
+                params={
+                    "chat_id": chat_id,
+                    "text": chat_id,
+                },
+            )
+
+            payload = response.json()
+            if not payload["ok"]:
+                logger.error(payload["description"])
+
+    def get_channel_id_from_message(self, message):
+        """See base class."""
+
+        # Since live posts aren't cleaned when they are added via a receiver,
+        # we make sure at this level that we return the correct types.
+        return str(message["chat"]["id"])
+
+    def get_message_id_from_message(self, message):
+        """Messages containing multiple images have a media_group_id attribute.
+        See base class.
+        """
+
+        msg_id = message.get("media_group_id") or message.get("message_id")
+        return str(msg_id)
+
+    def get_message_text(self, message):
+        """See base class."""
+
+        # Telegram parses the text of a message before sending it.
+        # The result can be found in the message's "entities".
+        return {
+            "text": message.get("text") or message.get("caption") or "",
+            "entities": message.get("entities", []),
+        }
+
+    def process_text(self, live_post, message_text):
+        """Use the message entities to convert links.
+        A raw link isn't converted by Telegram.
+        A link with a description is sent as a `text_link` entity.
+        See base class.
+        """
+
+        text = message_text["text"]
+        len_text = len(text)
+
+        cumulated_offset = 0
+        entities = message_text["entities"]
+        for entity in entities:
+            url = ""
+            offset = int(entity["offset"]) + cumulated_offset
+            length = int(entity["length"])
+            end = offset + length
+
+            if entity["type"] == "url":
+                url = description = text[offset:end]
+                if is_embed(url):
+                    # Check if this can match an embed block, if so no conversion happens.
```

Comment:
Hi Tidiane,

Could you give an example what you are trying to detect here? What case should be converted and what case shouldn't? Thanks :-)

Reply:
Hi Lucas, this shouldn't be converted for example:
```
"This post contains an embed.\n"
"https://www.youtube.com/watch?v=Cq3LOsf2kSY\n"
```
This however should be converted:
```
"https://www.youtube.com/watch?v=Cq3LOsf2kSY. Some content here.\n"
 "Some content here. https://www.youtube.com/watch?v=Cq3LOsf2kSY"
 ```

The idea is, if the embed url is the first line, or inside `\n` or the last line of the message, the `process_text` method will convert it into an embed.

If there is other text in the same line than the embed url, we convert it.

Reply:
Ah, because embeds are different than links to embeds of course :-) I propose a little note describing this to remind the reader!

Reply:
I guess the comment I put isn't clear.

Reply:
:-) Now I know what you mean, but it is often useful to give a small explanation why something is, because readers might not know or deduce it. So I think
```
# Check if this can match an embed block, if so no conversion happens.
```
Would be clearer like:
```
# Check if this can match an embed block, if so no conversion happens, because embeds should not become links.
```
:-)